recipes/zlib/1.2.11/conanfile.py

```diff
@@ -44,13 +38,14 @@ def package_id(self):
         del self.info.options.minizip
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        tools.get(**self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def _patch_sources(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
+        with tools.chdir(".."):  # FIXME: This need to go to parent folder is not very nice.
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch, base_path=self.source_folder)
```

Comment:
See https://github.com/conan-io/conan/pull/9361
`apply_conandata_patches` can be used to follow the layout automatically.